threatgrid/ctia
# Ensure pagination for fetching judgements of the feed

> Close advthreat/iroh#7281

- Add `ctia.stores.es.store/all-pages-iteration` function to iterate lazily over the result of ES query
- Ensure feed will not try to fetch more than `max_result_window` judgements or relationships at once

<a name="qa">[§](#qa)</a> QA
============================

1. Import a single indicator and a lot of judgements (more than 10000) related to that indicator via related relationships with `"source_ref": "%indicator-id%"`
2. Create a feed using indicator id from the result of the first step
3. Get a view of the feed `GET /ctia/feed/:feed-id/view?s=:secret`. Response should contain all related judgements and there are should be pagination headers (`X-Limit` and `X-Search_after`) present in the response.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: implement ctia.store/IPaginateableStore protocol for the store that needs pagination capabilities
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================
